src/components/Table.js

```diff
@@ -241,46 +241,72 @@ export default class Table extends React.Component<TableDefaultProps, TableProps
   }
 
   componentWillReceiveProps(newProps: TableProps) {
-    const { rows, noEmptySelection } = newProps;
+    const { rows, noEmptySelection, rowComparator, onSelect } = newProps;
     const { rows: prevRows } = this.props;
+    const { selectedIndices: prevSelectedIndices, activeRowIndex } = this.state;
 
     // If a rowComparator is specified, use it, otherwise, do a deep row comparison and reset selection if anything changed.
     const shouldUpdateRows = !isEqual(rows, prevRows);
-    const shouldResetSelection = rows.length !== prevRows.length
-      || (newProps.rowComparator && !isEqualWith(rows, prevRows, newProps.rowComparator))
-      || (!newProps.rowComparator && shouldUpdateRows);
+    let shouldResetSelection = (rowComparator && !isEqualWith(rows, prevRows, rowComparator))
+      || (!rowComparator && shouldUpdateRows);
 
     if (shouldResetSelection || shouldUpdateRows) {
+      const updatedSelectedRows = [];
+
       const sortedRows = rows && rows.length > 0
         ? rows.map((row, tableRowIndex) => {
-          return { ...row, tableRowIndex };
+          const updatedRow = { ...row, tableRowIndex };
+          if (prevSelectedIndices.has(tableRowIndex)) {
```

Comment:
In this loop, you're adding any previously selected rows to the updated selection if they're still in the list, right? Maybe call this out explicitly in a comment?